tests/integ/test_column_names.py

```diff
@@ -648,49 +763,89 @@ def test_binary_expression(session):
         df1["a"] / df1["a"],
         df1["a"] % df1["a"],
     )
-    assert (
-        [x.name for x in df3._output]
-        == df3.columns
-        == [
+    verify_column_result(
+        session,
+        df3,
+        [
             '"(""B"" = \'X\')"',
             '"(""B"" != \'X\')"',
             '"(""B"" > \'X\')"',
             '"(""B"" <= \'X\')"',
             '"EQUAL_NULL(""A"", ""B"")"',
-            '"((""B"" = \'X\') AND (""A"" = 1 :: INT))"',
-            '"((""B"" = \'X\') OR (""A"" = 1 :: INT))"',
-            '"BITAND(1 :: INT, ""A"")"',
-            '"BITOR(1 :: INT, ""A"")"',
-            '"BITXOR(1 :: INT, ""A"")"',
-            '"POWER(""A"", 2 :: INT)"',
+            '"((""B"" = \'X\') AND (""A"" = 1))"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"((""B"" = \'X\') AND (""A"" = 1 :: INT))"',
+            '"((""B"" = \'X\') OR (""A"" = 1))"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"((""B"" = \'X\') OR (""A"" = 1 :: INT))"',
+            '"BITAND(1, ""A"")"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"BITAND(1 :: INT, ""A"")"',
+            '"BITOR(1, ""A"")"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"BITOR(1 :: INT, ""A"")"',
+            '"BITXOR(1, ""A"")"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"BITXOR(1 :: INT, ""A"")"',
+            '"POWER(""A"", 2)"'
+            if session.eliminate_numeric_sql_value_cast_enabled
+            else '"POWER(""A"", 2 :: INT)"',
             '"(""A"" + ""A"")"',
             '"(""A"" - ""A"")"',
             '"(""A"" * ""A"")"',
             '"(""A"" / ""A"")"',
             '"(""A"" % ""A"")"',
-        ]
+        ],
+        [
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            BooleanType(),
+            LongType() if not local_testing_mode else BooleanType(),
```

Comment:
@sfc-gh-aling @sfc-gh-stan The bitwise operator seems returning Boolean type for local testing mode here, is that expected?

Reply:
No it's not. I know why though. This is likely the `ColumnEmulator` doesn't have a numeric `dtype` when the bitwise operation happened. If the dtype is `object`, the return value will be `boolean`. That's why we have https://github.com/snowflakedb/snowpark-python/blob/8a4276387e09947d55139fac6a1d6e3590f3f891/src/snowflake/snowpark/mock/_util.py#L215

Reply:
Oh, do we have a plan to fix that? is here a ticket to track this ?

Reply:
I don't think this is a known issue. @sfc-gh-aling can you help unblock Yun here?

Reply:
I don't think i am necessary blocking here, I can create a ticket for you to look into this if this is not a known issue, and left a comment on the test

Reply:
Let's create a followup ticket for this. cc: @sfc-gh-jrose @sfc-gh-aling

Reply:
created a ticket here https://snowflakecomputing.atlassian.net/browse/SNOW-1524633.  @sfc-gh-aling i assigned it to your for now, feel free to triage it.

Reply:
also added the ticket number in comment